docs/element-definitions.md

```diff
@@ -1,1419 +1,1623 @@
 
-#### Example
+#### Examples
 
-Given an API description with following layout:
+Type Element matching only property lists (JSON `object`).
 
-- Group
-    - Copy "Lorem Ipsum"
-    - Resource "A"
-    - Resource "B"
-    - Copy "Dolor Sit Amet"
+```json
+
+{
+  "element": "object"
+}
+
+```
+
+Type Element matching only a specific property list instance (JSON `{"foo": false, "bar": "fun"}`).
 
 ```json
+
 {
-  "element": "category",
+  "element": "object",
+  "attributes": {
+    "typeAttributes": {
+      "element": "array",
+      "content": [
+        {
+          "element": "string",
+          "content": "fixed"
+        }
+      ]
+    }
+  },
   "content": [
     {
-      "element": "copy",
-      "content": "Lorem Ipsum"
+      "element": "member",
+      "content": {
+        "key": {
+          "element": "string",
+          "content": "foo"
+        },
+        "value": {
+          "element": "boolean"
+          "content": false
+        }
+      }
     },
     {
-      "element": "resource"
-    },
+      "element": "member",
+      "content": {
+        "key": {
+          "element": "string",
+          "content": "foo"
+        },
+        "value": {
+          "element": "string"
+          "content": "fun"
+        }
+      }
+    }
+  ]
+}
+
+```
+
+Type Element matching only a property list with key "foo" of value type `boolean` and with the key "bar" of value type `string` (JSON `{"foo": false, "bar": "fun"}`, `{"foo": true, "bar": ""}`).
+
+```json
+
+{
+  "element": "object",
+  "attributes": {
+    "typeAttributes": {
+      "element": "array",
+      "content": [
+        {
+          "element": "string",
+          "content": "fixedType"
+        }
+      ]
+    }
+  },
+  "content": [
     {
-      "element": "resource"
+      "element": "member",
+      "content": {
+        "key": {
+          "element": "string",
+          "content": "foo"
+        },
+        "value": {
+          "element": "boolean",
+          "content": false
+        }
+      }
     },
     {
-      "element": "copy",
-      "content": "Dolor Sit Amet"
+      "element": "member",
+      "content": {
+        "key": {
+          "element": "string",
+          "content": "bar"
+        },
+        "value": {
+          "element": "string",
+          "content": "fun"
+        }
+      }
     }
   ]
 }
-```
 
-### Protocol-specific Elements
+```
 
-#### HTTP Transaction (Base API Element)
+---
 
-Example of an HTTP Transaction. The example's content consist of a request-response
-message pair. A transaction example MUST contain exactly one HTTP request and one HTTP response message.
+### Enum Element
 
-##### Properties
+Type with the domain of any of the types listed in the `enumerations` attribute. Also called tagged union or Σ-type.
```

Comment:
Type with the domain of the union of values typed by Elements in the `enumerations` attribute.